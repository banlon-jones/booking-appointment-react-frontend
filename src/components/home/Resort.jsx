import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Image from 'react-bootstrap/Image';
import classes from './Resorts.module.css';

const Resort = (props) => {
  const { resort } = props;
  const detailsUrl = `/resorts/${resort.id}`;

  return (
    <Link to={detailsUrl}>
      <Image roundedCircle src={resort.image} alt={resort.name} />
      <h2 className={classes.resort_name}>{resort.name}</h2>
    </Link>
  );
};

Resort.propTypes = {
  resort: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
  }).isRequired,
};

export default Resort;
