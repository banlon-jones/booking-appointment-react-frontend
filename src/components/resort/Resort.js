/* eslint-disable comma-dangle */
/* eslint-disable no-unused-vars */

import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { BsChevronRight } from 'react-icons/bs';
import { useNavigate, useParams } from 'react-router-dom';
import {
  selectResort,
  getResort,
} from '../../store/resort-details/resortDetail';
import resortClass from './Resort.module.css';

function Resort() {
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const { id } = useParams();

  useEffect(() => {
    dispatch(getResort(id));
  }, []);

  const user = useSelector((state) => state.user);
  const resortDetails = useSelector(selectResort);
  return (
    <div className={resortClass.resort_container}>
      <div className={resortClass.inner_container}>
        <div className={resortClass.image_container}>
          <img src={resortDetails.image} alt="logo" />
        </div>

        <div className={resortClass.resort_details_container}>
          <h1>{resortDetails.name}</h1>
          <ul>
            <li>
              Country:
              <span>{resortDetails.country}</span>
            </li>
            <li>
              City:
              <span>{resortDetails.city}</span>
            </li>
            <li>
              Description:
              <p>{resortDetails.description}</p>
            </li>
            <li>
              Cost:
              <span>{`$${resortDetails.cost}`}</span>
            </li>
          </ul>
          <div className={resortClass.button_main_container}>
            <button
              type="button"
              className={resortClass.button_container}
              onClick={() => {
                navigate(`/resorts/${resortDetails.id}`);
              }}
            >
              Reserve
              <BsChevronRight className={resortClass.icon} />
            </button>
            <button
              type="button"
              className={resortClass.button_container}
              style={{ display: user.role === 'admin' ? 'block' : 'none' }}
              onClick={() => {
                navigate(`/resort-update/${resortDetails.id}`);
              }}
            >
              Update
              <BsChevronRight className={resortClass.icon} />
            </button>
          </div>
        </div>
      </div>
    </div>
  );
}

export default Resort;
