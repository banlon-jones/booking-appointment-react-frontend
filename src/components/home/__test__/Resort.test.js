import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BrowserRouter as Router } from 'react-router-dom';

import Resort from '../Resort';

describe('Resorts snapshots', () => {
  const resort = {
    id: 1,
    name: 'Resort Name',
    image: 'image_url',
  };
  const initialState = { resorts: [] };
  const mockStore = configureStore();
  let store;

  it('matches Resort snapshot', () => {
    store = mockStore(initialState);
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Resort resort={resort} />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('Page rendering', () => {
    beforeEach(() => {
      render(
        <Provider store={store}>
          <Router>
            <Resort resort={resort} />
          </Router>
        </Provider>,
      );
    });

    test('renders the Resorts page heading', () => {
      expect(screen.getByRole('heading')).toHaveTextContent(/Resort Name/);
    });

    test('renders image in the ResortsToDelete page', () => {
      expect(screen.getAllByRole('img')[0].src).toContain('image_url');
    });
  });
});
