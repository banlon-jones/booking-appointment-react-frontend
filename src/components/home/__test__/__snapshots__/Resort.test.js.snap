// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resorts snapshots matches Resort snapshot 1`] = `
<a
  href="/resorts/1"
  onClick={[Function]}
>
  <img
    alt="Resort Name"
    className="rounded-circle"
    src="image_url"
  />
  <h2
    className="resort_name"
  >
    Resort Name
  </h2>
</a>
`;
